app/javascript/components/character-count.js

```diff
@@ -0,0 +1,187 @@
+function CharacterCount ($module) {
+  this.$module = $module
+}
+
+CharacterCount.prototype.defaults = {
+  characterCountAttribute: 'data-maxlength',
+  wordCountAttribute: 'data-maxwords'
+}
+
+// Initialize component
+CharacterCount.prototype.init = function (options) {
+  // Check for module
+  var $module = this.$module
+  if (!$module) {
+    return
+  }
+
+  // Read options set using dataset ('data-' values) and merge them with options
+  var dataset = this.getDataset($module)
+  this.options = this.merge(options || {}, dataset)
+
+  // Determine the limit attribute (characters or words)
+  var countAttribute = (this.options.maxwords) ? this.defaults.wordCountAttribute : this.defaults.characterCountAttribute
+
+  // Set the element limit
+  if ($module.getAttribute) {
+    this.maxLength = $module.getAttribute(countAttribute)
+  }
+
+  // Generate and reference message
+  var boundCreateCountMessage = this.createCountMessage.bind(this)
+  this.countMessage = boundCreateCountMessage()
+
+  // If there's a maximum length defined and the count message was successfully applied
+  if (this.maxLength && this.countMessage) {
+    // Replace maxlength attribute with data-maxlength to avoid hard limit
+    $module.setAttribute('maxlength', '')
+    $module.setAttribute('data-maxlength', $module.maxLength)
+
+    // Bind event changes to the textarea
+    var boundChangeEvents = this.bindChangeEvents.bind(this)
+    boundChangeEvents()
+
+    // Update count message
+    var boundUpdateCountMessage = this.updateCountMessage.bind(this)
+    boundUpdateCountMessage()
+  }
+}
+
+// Fills in default values
+CharacterCount.prototype.merge = function (obj) {
+  for (var i = 1; i < arguments.length; i++) {
+    var def = arguments[i]
+    for (var n in def) {
+      if (obj[n] === undefined) obj[n] = def[n]
+    }
+  }
+  return obj
+}
+
+CharacterCount.prototype.getDataset = function (element) {
```

Comment:
Again, would it be easier to just fish these out manually if there aren’t many?

Reply:
Or perhaps extract this to somewhere more general as it looks like it could be generally useful :-)